### PR TITLE
Feat: restructure the Hillla documentation and add the React section

### LINKED ISSUE #1840: Data entry components article
Describe React component wrappers for the following Vaadin components with at least one code example per component, in priority order:

```
vaadin-text-field / number / password / email / integer
vaadin-checkbox / checkbox-group
vaadin-radio-button / radio-group
vaadin-combo-box
vaadin-date-picker / time / date-time
vaadin-select
vaadin-list-box / item
vaadin-text-area
vaadin-upload
vaadin-custom-field
```

### LINKED ISSUE #1841: Basics tutorial for React
Create a React based tutorial similar with the lit based [Hilla Basics Tutorial] (https://hilla.dev/docs/tutorials/basics-tutorial)

### LINKED ISSUE #1842: Layout components article
Describe React component wrappers for the following Vaadin components with at least one code example per component, in priority order:

```
vaadin-app-layout / drawer-toggle
vaadin-form-layout
vaadin-split-layout
vaadin-horizontal-layout / vertical
```

### LINKED ISSUE #1843: Interaction components article
Describe React component wrappers for the following Vaadin components with at least one code example per component, in priority order:

```
vaadin-button
vaadin-grid
vaadin-virtual-list
vaadin-dialog / confirm-dialog
vaadin-tabs / tab
vaadin-details
vaadin-scroller
```

### PR DESCRIPTION
**Pull request by jouni**
**Contains a lot of commits: squashing this one would be problematic.** It is probably better to use fast-forward or a merge commit.

Restructure Lit documentation content to tutorials, guides, and reference.
Add a section for React with the basics tutorial and documentation for a selection of components.

Closes #1840 
Closes #1841 
Closes #1842 
Closes #1843 

*Note*: the preview build does not look like the finished product, there are some updates in DSP/docs-app that are not used in the Amplify preview. ~~And the preview build will also fail because of all of the broken cross references.~~

### CODE REVIEW

File: articles/lit/start/faq.adoc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+It seamlessly integrates a reactive https://lit.dev/[Lit] TypeScript front end with a https://spring.io/projects/spring-boot[Spring Boot] back end.
+endif::[]
+ifdef::react[]
+It seamlessly integrates a https://reactjs.org/[React] TypeScript front end with a https://spring.io/projects/spring-boot[Spring Boot] back end.
+endif::[]
+
+Hilla is designed to be simple enough for small utilities, but robust enough to build complex, enterprise-grade applications.
+
+The automatic type generation ensures that your front end and back end always stay in sync and that you catch mistakes at build time, not runtime.
+
```
Comment (tarekoraby) [inline articles/lit/start/faq.adoc:24]:
```suggestion

== What are the benefits of using Hilla?

Hilla simplifies the process of creating business applications. It:

- saves development time by simplifying communication between the front end and Java back end, eliminating the need to create a REST API;
- enforces end-to-end type-safety, identifying errors during the build process, when they are faster to fix, and ensuring they don’t make it into production;
- reduces the time to build a UI with 45+ ready-made components plus a fully-customizable theme, all maintained by Vaadin;
- simplifies security with endpoints that are secured by default and validation that ensures data integrity across the server and client;
- gets you started quickly with a zero-configuration toolchain for building both your front and back end;

```

**Comment by tarekoraby** [comment]
A small nitpick, I feel it would be better to remove the icon next to the "Reactive endpoints" chapter. The presence of it makes the chapter seems extra important compared to all the others, which it is not. 

image

**Comment by jouni** [comment]
Yeah, it was a bit of a test if we should promote experimental features more in the navigation, similarly as we do with commercial stuff in Vaadin docs.

File: articles/lit/start/basics/index.adoc [hunk 2]
```diff
@@ -0,0 +1,975 @@
...
+You should install the following plugins in VS Code for an optimal development experience:
+
+* https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack[Java Extension Pack]
+ifdef::lit[]
+* https://marketplace.visualstudio.com/items?itemName=runem.lit-plugin[lit-plugin]
+endif::[]
+* https://marketplace.visualstudio.com/items?itemName=Pivotal.vscode-spring-boot[Spring Boot Tools]
++
+ifdef::react[]
+* https://plugins.jetbrains.com/plugin/9997-styled-components--styled-jsx[Styled Components & Styled JSX]
```
Comment (cromoteca) [inline articles/lit/start/basics/index.adoc:72]: This is a JetBrains plugin, not related to VS Code.

**Comment by starg09** [comment]
> Yeah, it was a bit of a test if we should promote experimental features more in the navigation, similarly as we do with commercial stuff in Vaadin docs. I assume we want users to try them out, but perhaps this is indeed too prominent to distract new users.

Def would be nice to get some indicator for these. Maybe a badge in that tone of yellow with "Exp." or "WIP"?

**Review by cromoteca — APPROVED** [review]

**Pull request merged**